.github/workflows/ci-workflow.yml

```diff
@@ -12,55 +12,59 @@ on:
     - develop
 
 jobs:
-  job_build_debug_nano_S:
-    name: Build debug NanoS
-    runs-on: ubuntu-latest
-
-    container:
-      image: ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder:latest
-
-    steps:
-      - name: Clone
-        uses: actions/checkout@v2
-
-      - name: Build
-        run: |
-          make DEBUG=1 TESTING=1 TEST_PUBLIC_KEY=1
-          cp bin/app.elf exchange_nanos.elf
+  build_sideloaded_applications:
+    strategy:
+      matrix:
+        sdk:
+        - path: $NANOS_SDK
+          name: nanos
+        - path: $NANOX_SDK
+          name: nanox
+        - path: $NANOSP_SDK
+          name: nanosp
+        coin:
+        - name: stellar
+          repo: app-stellar
+        - name: tezos
+          repo: app-tezos
+        - name: xrp
+          repo: app-xrp
+        - name: ethereum
+          repo: app-ethereum
+        - name: litecoin
+          repo: app-bitcoin-new
+        - name: bitcoin
+          repo: app-bitcoin-new
 
-      - name: Upload app binary
-        uses: actions/upload-artifact@v2
-        with:
-          name: exchange
-          path: ./exchange_nanos.elf
-          if-no-files-found: error
-
-  job_build_debug_nano_X:
-    name: Build debug NanoX
     runs-on: ubuntu-latest
-
     container:
       image: ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder:latest
 
     steps:
-      - name: Clone
-        uses: actions/checkout@v2
+      - name: Clone ${{ matrix.coin.repo }} for coin ${{ matrix.coin.name }}
+        uses: actions/checkout@v3
+        with:
+          repository: LedgerHQ/${{ matrix.coin.repo }}
+          ref: develop
 
-      - name: Build
+      - name: Build in ${{ matrix.coin.repo }} coin ${{ matrix.coin.name }} for SDK ${{ matrix.sdk.name }}
         run: |
-          make BOLOS_SDK=$NANOX_SDK DEBUG=1 TESTING=1 TEST_PUBLIC_KEY=1
-          cp bin/app.elf exchange_nanox.elf
+          make clean ; COIN=${{ matrix.coin.name }} BOLOS_SDK=${{ matrix.sdk.path }} make
+          cp bin/app.elf ${{ matrix.coin.name }}_${{ matrix.sdk.name }}.elf
 
-      - name: Upload app binary
+      - name: Upload ${{ matrix.coin.name }} binaries
         uses: actions/upload-artifact@v2
         with:
-          name: exchange
-          path: ./exchange_nanox.elf
+          name: ${{ matrix.coin.name }}_binaries
```

Comment:
Not sure how this will work, as the matrix will create 3 jobs, each with it's own workdir, and each uploading the artifact with the same name ?
Maybe for clarity :
name: ${{ matrix.coin.name }}_${{ matrix.sdk.name }}_binaries

Reply:
Uploading with the same artifact name adds (or overwrite) to the files contained in the artifact archive. This is handy to only download once when you need lots of artifcacts.
In this case, as the artifacts are renamed, there is no overwritting, every variants of a coin app are upload in the same archive.
I would argue all variants of all coin apps could be stored in the same artifact. No need for segregations on these.